### PR TITLE
Select

### PR DESCRIPTION
**Pull request by cvybhu**
Adds select example to `main.c`

### CODE REVIEW

**Comment by cvybhu** [comment]
Fails with linking error ;-;
```
[ 10%] Creating directories for 'scylla-rust-driver'
[ 20%] No download step for 'scylla-rust-driver'
[ 30%] No update step for 'scylla-rust-driver'
[ 40%] No patch step for 'scylla-rust-driver'
[ 50%] No configure step for 'scylla-rust-driver'
[ 60%] Performing build step for 'scylla-rust-driver'
-- scylla-rust-driver build command succeeded.  See also /home/jancio/code/cpp-rust-driver/bui/Rust/src/scylla-rust-driver-stamp/scylla-rust-driver-build-*.log
[ 70%] No install step for 'scylla-rust-driver'
[ 80%] Completed 'scylla-rust-driver'
[ 80%] Built target scylla-rust-driver
[ 90%] Building C object CMakeFiles/main.dir/src/main.c.o
[100%] Linking C executable main
/usr/bin/ld: CMakeFiles/main.dir/src/main.c.o: in function `main':
/home/jancio/code/cpp-rust-driver/src/main.c:33: undefined reference to `cass_future_get_result'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:34: undefined reference to `cass_iterator_from_result'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:36: undefined reference to `cass_iterator_get_row'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:39: undefined reference to `cass_row_get_column'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:39: undefined reference to `cass_value_get_int32'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:40: undefined reference to `cass_row_get_column'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:40: undefined reference to `cass_value_get_int32'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:41: undefined reference to `cass_row_get_column'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:41: undefined reference to `cass_value_get_int32'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:35: undefined reference to `cass_iterator_next'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:45: undefined reference to `cass_iterator_free'
/usr/bin/ld: /home/jancio/code/cpp-rust-driver/src/main.c:46: undefined reference to `cass_result_free'
collect2: error: ld returned 1 exit status
distcc[298623] ERROR: compile (null) on localhost failed
make[2]: *** [CMakeFiles/main.dir/build.make:98: main] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

**Comment by jul-stas** [comment]
I thought we'll start with the simpler `cass_result_first_row()` but this is even better :+1:

**Comment by cvybhu** [comment]
Okay, missed `[no_mangle]`, everything works now

File: scylla-rust-wrapper/src/query_result.rs [hunk 1]
```diff
@@ -39,23 +39,36 @@ pub unsafe extern "C" fn cass_iterator_free(iterator: *mut CassIterator) {
...
 pub unsafe extern "C" fn cass_iterator_next(iterator: *mut CassIterator) -> cass_bool_t {
     let iter: &mut CassIterator = ptr_to_ref_mut(iterator);
-    iter.position += 1;
 
-    match &iter.result.rows {
-        Some(rs) => (iter.position < rs.len()) as cass_bool_t,
+    let new_pos: usize = match iter.position {
+        Some(prev_pos) => prev_pos + 1,
         None => 0,
+    };
```
Comment (piodul) [inline scylla-rust-wrapper/src/query_result.rs:53]: Suggestion:

```rust
let new_pos: usize = iter.position.map_or(0, |x| x + 1);
```

**Review by piodul — APPROVED** [review]

File: scylla-rust-wrapper/src/query_result.rs (hunk 1 above)
Comment (cvybhu) [reply scylla-rust-wrapper/src/query_result.rs:53]: I don't know, the current code seems easier to understand.

**Pull request merged**